File: Php/atm.php
<?php
$nama = readline("Masukkan nama Anda: ");

// validasi nama 
while (true){
    echo "Hallo, $nama! Apakah nama kamu sudah benar? (y/n): ";
    $konfirmasi = trim(fgets(STDIN));

    if ($konfirmasi == 'y' || $konfirmasi == 'Y') {
        break;
    } elseif ($konfirmasi == 'n' || $konfirmasi == 'N') {
        $nama = readline("Masukkan nama Anda kembali: ");
    } else {
        echo "Input tidak valid. Silakan masukkan 'y' untuk ya atau 'n' untuk tidak.\n";
    }
}

$saldo = readline("Masukkan saldo awal Anda: Rp ");

echo "Selamat datang di ATM, $nama!\n";
echo "Saldo awal Anda: Rp $saldo\n";

echo "Pilih operasi yang ingin Anda lakukan:\n";
echo "1. Cek Saldo\n";
echo "2. Tarik Tunai\n";
echo "3. Transfer\n";
echo "4. Keluar\n";

$pilihan = readline("Masukkan pilihan Anda (1-4): ");

switch ($pilihan) {
    case '1':
        echo "Saldo Anda saat ini: Rp $saldo\n";
        break;

    case '2':
        $jumlah = readline("Masukkan jumlah uang yang ingin Anda tarik: Rp ");
        if ($jumlah > $saldo) {
            echo "Saldo tidak mencukupi. Saldo Anda saat ini: Rp $saldo\n";
        } else {
            $saldo -= $jumlah;
            echo "Uang Anda berhasil ditarik. Saldo Anda saat ini: Rp $saldo\n";
        }
        break;

    case '3':
        $rekening = readline("Masukkan nomor rekening tujuan: ");
        $jumlah = readline("Masukkan jumlah uang yang ingin Anda transfer: Rp ");
        if ($jumlah > $saldo) {
            echo "Saldo tidak mencukupi. Saldo Anda saat ini: Rp $saldo\n";
        } else {
            $saldo -= $jumlah;
            echo "Uang sebesar Rp $jumlah berhasil ditransfer ke rekening $rekening. Saldo Anda saat ini: Rp $saldo\n";
        }
        break;

    case '4':
        echo "Terima kasih telah menggunakan ATM kami. Sampai jumpa!\n";
        break;
        
    default:
        echo "Pilihan tidak valid. Silakan masukkan angka 1-4.\n";
        break;
}
?>

File: Java/atm.java
import java.util.Scanner;

public class atm {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Masukkan nama Anda: ");
        String nama = input.nextLine();

        while (true) {
            System.out.print("Hallo, " + nama + "! Apakah nama kamu sudah benar? (y/n): ");
            String konfirmasi = input.nextLine();

            if (konfirmasi.equalsIgnoreCase("y")) {
                break;
            } else if (konfirmasi.equalsIgnoreCase("n")) {
                System.out.print("Masukkan nama Anda kembali: ");
                nama = input.nextLine();
            } else {
                System.out.println("Input tidak valid. Silakan masukkan 'y' atau 'n'.");
            }
        }

        System.out.print("Masukkan saldo awal Anda: Rp ");
        double saldo = input.nextDouble();
        input.nextLine(); 

        System.out.println("\nSelamat datang di ATM, " + nama + "!");
        System.out.println("Saldo awal Anda: Rp " + saldo);

        System.out.println("\nPilih operasi yang ingin Anda lakukan:");
        System.out.println("1. Cek Saldo");
        System.out.println("2. Tarik Tunai");
        System.out.println("3. Transfer");
        System.out.println("4. Keluar");

        System.out.print("Masukkan pilihan Anda (1-4): ");
        int pilihan = input.nextInt();
        input.nextLine();

        switch (pilihan) {
            case 1:
                System.out.println("Saldo Anda saat ini: Rp " + saldo);
                break;

            case 2:
                System.out.print("Masukkan jumlah uang yang ingin Anda tarik: Rp ");
                double tarik = input.nextDouble();
                if (tarik > saldo) {
                    System.out.println("Saldo tidak mencukupi. Saldo Anda: Rp " + saldo);
                } else {
                    saldo -= tarik;
                    System.out.println("Penarikan berhasil! Saldo Anda saat ini: Rp " + saldo);
                }
                break;

            case 3:
                System.out.print("Masukkan nomor rekening tujuan: ");
                String rekening = input.next();

                System.out.print("Masukkan jumlah uang yang ingin Anda transfer: Rp ");
                double transfer = input.nextDouble();

                if (transfer > saldo) {
                    System.out.println("Saldo tidak mencukupi. Saldo Anda: Rp " + saldo);
                } else {
                    saldo -= transfer;
                    System.out.println("Transfer Rp " + transfer + " ke rekening " + rekening + " berhasil.");
                    System.out.println("Saldo Anda saat ini: Rp " + saldo);
                }
                break;

            case 4:
                System.out.println("Terima kasih telah menggunakan ATM. Sampai jumpa!");
                break;

            default:
                System.out.println("Pilihan tidak valid. Masukkan angka 1-4.");
                break;
        }

        input.close();
    }
}
